merkle-tree/concurrent/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+use std::{cmp::Ordering, marker::PhantomData};
+
+use light_hasher::{errors::HasherError, Hasher};
+
+pub mod changelog;
+pub mod hash;
+
+use crate::{
+    changelog::ChangelogEntry,
+    hash::{compute_parent_node, full_proof, is_valid_proof},
+};
+
+pub type MerkleProof<const MAX_HEIGHT: usize> = [[u8; 32]; MAX_HEIGHT];
+
+#[repr(C)]
+pub struct ConcurrentMerkleTree<H, const MAX_HEIGHT: usize, const MAX_ROOTS: usize>
+where
+    H: Hasher,
+{
+    /// History of Merkle proofs.
+    pub changelog: [ChangelogEntry<MAX_HEIGHT>; MAX_ROOTS],
```

Comment:
I think there should be a separate const generic for eg MAX_ENTRIES to set the size of the changelog